### PR TITLE
Enhance search in Checkboxes (Tree) interface

### LINKED ISSUE #18918: v-checkbox-tree search is not accent sensitive
### Describe the Bug

Hello Directus Team, 

I was using the `v-checkbox-tree` component and I've noticed that it's not accent sensitive in search. While doing some research, I found that you have a package installed `diacritics` with a function called `removeDiacritics` which is used by the `v-highlight component`. This component is used within `v-checkbox-tree`.

image
[reference.](https://github.com/directus/directus/blob/10215abdd0575bc92019381170e362b98388d6cd/app/src/components/v-highlight.vue#L32)

In my understanding, what I see as a bug is that `removeDiacritics` is not used in the search function of `v-checkbox-tree`. I believe that a simple change in the `searchChoices` function by adding `removeDiacritics` should be sufficient.

image
[reference.](https://github.com/directus/directus/blob/10215abdd0575bc92019381170e362b98388d6cd/app/src/components/v-checkbox-tree/v-checkbox-tree.vue#LL123C3-L123C3)

Thank you very much.
Sincerely,
Binpar Team

### To Reproduce

It is a code review, so there aren't many steps to follow. 

To replicate this issue, you need to have a `v-checkbox-tree` component with a search input and enter a text with accents in the search input. In this case, you can try with "Iлｔèｒｎåｔïｏｎɑｌíƶａｔï߀ԉ", which is the text used by `diacritics` in their examples.

### Directus Version

v10.3.0

### Hosting Strategy

Self-Hosted (Docker Image)

### PR DESCRIPTION
**Pull request by gitstart**
Fixes #18918

Demo: https://www.loom.com/share/8f5a711e0ade4c1ba03c6f30d8ec15e2

### CODE REVIEW

File: .changeset/dirty-shrimps-laugh.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@directus/app': patch
+---
+
+Make v-checkbox-tree search accent sensitive
```
Comment (paescuj) [inline .changeset/dirty-shrimps-laugh.md:5]:
```suggestion
Enabled accent sensitive search in checkbox tree component 
```

**Review by paescuj — APPROVED** [review]

**Review by DanielBiegler — CHANGES_REQUESTED** [review]
I think the highlighting change broke something because on 0baab84c36d0a91fe4dca13437369eddc8a02dd4 it works:

image

But then on 349db6ec7d894af548e3ad6237a697b3d8306641 it breaks:

image

Also is this behaviour a separate issue or related?:

[Screencast from 18.08.2023 19:18:34.webm](https://github.com/directus/directus/assets/14810858/1848d27f-bee2-4b18-8165-55b192979149)

**Comment by paescuj** [comment]
> I think the highlighting change broke something because on [0baab84](https://github.com/directus/directus/commit/0baab84c36d0a91fe4dca13437369eddc8a02dd4) it works:
> 
> image
> 
> But then on [349db6e](https://github.com/directus/directus/commit/349db6ec7d894af548e3ad6237a697b3d8306641) it breaks:
> 
> image

Looking into it 👍 

> Also is this behaviour a separate issue or related?:
> Screencast.from.18.08.2023.19.18.34.webm

I think that has already been the case before.

File: app/package.json [hunk 2]
```diff
@@ -138,6 +136,7 @@
 		"micromustache": "8.0.3",
 		"mime": "3.0.0",
 		"mitt": "3.0.0",
+		"modern-diacritics": "2.3.1",
```
Comment (rijkvanzanten) [inline app/package.json:139]: I have my doubts about this packages. No downloads, no indication of maintenance. I know the original diacritics hasn't been updated in what feels like centuries, but it still has a huge amount of people checking/relying on it 🤔

Comment (paescuj) [reply app/package.json:139]: >  I know the original diacritics hasn't been updated in what feels like centuries, but it still has a huge amount of people checking/relying on it 🤔

That's sums up the problem of the Node.js ecosystem quite well, if you ask me 🤣 People just don't care about outdated packages, so I'm not sure whether this number should be that decisive. I took a glance at `modern-diacritics` and it made a solid impression to me. If for example a security bug should appear, I also think that the chances are higher that the newer package will receive a fix than the package that has not received any updates for 7 years.
But I get your doubts, I'm always checking number of downloads too!

Screenshot 2023-08-18 at 20-16-17 diacritics vs fold-to-ascii vs modern-diacritics vs normalize-diacritics npm trends
https://npmtrends.com/diacritics-vs-fold-to-ascii-vs-modern-diacritics-vs-normalize-diacritics

We could also take a look at [`normalize-diacritics`](https://www.npmjs.com/package/normalize-diacritics
).

Let me know what you prefer!

**Comment by DanielBiegler** [comment]
@paescuj my brother in christ, go to sleep lmaooo

**Comment by paescuj** [comment]
Love the name of the changeset btw. 🦐😆

**Review by DanielBiegler — APPROVED** [review]
Looks really nice! 

[Screencast from 21.08.2023 12:11:26.webm](https://github.com/directus/directus/assets/14810858/6b4b3f9d-dac7-4e42-a759-39cd36f68d42)

File: app/package.json (hunk 2 above)
Comment (rijkvanzanten) [reply app/package.json:139]: A third option is: this library really doesn't do anything from a find and replace on a list of hardcoded diacritics characters. Almost wondering if it makes more sense to just roll that ourselves and remove the dependency on third party software altogether

Comment (paescuj) [reply app/package.json:139]: Right, also played with this thought! 👍

Comment (paescuj) [reply app/package.json]: Will be addressed in a separate PR 👍

**Pull request merged**